File: Domain/AL/Services/Lifeforms/Queries/GetSingleAnimal/AnimalDetails.cs
using Domain.DL.Models.LifeformModels;
using Shared.CQRS.Queries;
using System.Linq.Expressions;

namespace Domain.AL.Services.Lifeforms.Queries.GetSingleAnimal;
public class AnimalDetails : BaseReadModel
{
    public string Species { get; private set; }
    public bool IsBird { get; private set; }
    public byte MaximumOffspring { get; private set; }
    public byte MinimumOffspring { get; private set; }
    public IEnumerable<int> MessageIds { get; private set; }

    public AnimalDetails(string species, bool isBird, byte maxOffspring, byte minOffspring, IEnumerable<int> messageIds)
    {
        Species = species;
        IsBird = isBird;
        MaximumOffspring = maxOffspring;
        MinimumOffspring = minOffspring;
        MessageIds = messageIds;
    }
}

public class AnimalDetailsQuery : BaseQuery<Animalia, AnimalDetails>
{
    public override Expression<Func<Animalia, AnimalDetails>> Map()
    {
        return e => new(e.Species, e.IsBird, e.MaximumOffspringsPerMating, e.MinimumOffspringsPerMating, e.Messages.Select(x => x.MessageMessageId));
    }
}

File: CrossplantformApplication/lifeform_watcher/lib/models/lifeforms/animal_details.dart
class AnimalDetails {
  final String species;
  final bool isBird;
  final int maximumOffspring;
  final int minimumOffspring;
  final List<int> messageIds;

  const AnimalDetails(
      {required this.species,
      required this.isBird,
      required this.maximumOffspring,
      required this.minimumOffspring,
      required this.messageIds});

  factory AnimalDetails.fromJson(Map<String, dynamic> json) {
    return AnimalDetails(
        species: json['species'],
        isBird: json['isBird'],
        maximumOffspring: json['maximumOffspring'],
        minimumOffspring: json['minimumOffspring'],
        messageIds: json['messageIds']);
  }
}
